system/services/routeHandlers/AssetRouteHandler.cfc

```diff
@@ -43,24 +43,49 @@ component implements="iRouteHandler" singleton=true {
 	}
 
 	public string function build( required struct buildArgs, required any event ) output=false {
-		var assetId    = buildArgs.assetId    ?: ""
-		var derivative = buildArgs.derivative ?: ""
-		var versionId  = buildArgs.versionId  ?: ""
-		var trashed    = IsBoolean( buildArgs.trashed ?: "" ) && buildArgs.trashed;
-		var link       = "";
+		var assetId       = buildArgs.assetId       ?: "";
+		var derivative    = buildArgs.derivative    ?: "";
+		var versionId     = buildArgs.versionId     ?: "";
+		var trackDownload = buildArgs.trackDownload ?: false;
+		var trashed       = IsBoolean( buildArgs.trashed ?: "" ) && buildArgs.trashed;
+		var link          = "";
+		var assetType     = _getAssetManagerService().getAsset(
+			  id           = assetId
+			, selectFields = [ "asset_type" ]
+		).asset_type ?: "";
+
+		if ( !isEmpty( assetType ) ) {
+			trackDownload = !structKeyExists( _getAssetImageType(), assetType );
```

Comment:
Rather than this, we could/should perhaps add explicit settings for asset types in `settings.assetmanager.types.{group}.{type}`. Currently we have, for example:

```cfc
types.document.pdf  = { serveAsAttachment=true, mimeType="application/pdf"  };
```

We could allow:

```cfc
types.document.pdf  = { serveAsAttachment=true, mimeType="application/pdf", trackDownloads=true  };
```

And we could default this to the same value as `serveAsAttachment`. Then in the logic here:

```cfc
assetType = assetManagerService.getAssetType( name=assetType );
trackDownload = IsTrue( assetType.trackDownloads ?: ( assetType.serveAsAttachment ) );
```